x/ccv/provider/keeper/hooks.go

```diff
@@ -74,15 +74,14 @@ func ValidatorConsensusKeyInUse(k *Keeper, ctx sdk.Context, valAddr sdk.ValAddre
 
 	inUse := false
 
-	// Search over all consumer keys which have been assigned in order to
-	// check if the validator being added is, or was, a consumer chain validator
-	k.IterateAllValidatorsByConsumerAddr(ctx, func(_ string, consumerAddr sdk.ConsAddress, _ sdk.ConsAddress) (stop bool) {
-		if consumerAddr.Equals(consensusAddr) {
+	// TODO JEHAN: Stopping iteration here (IMO, this doesn't matter since iteration will never be
+	// stopped early in normal use. If this is a problem, then it needs to be solved with an index on ConsumerAddr)
+	for _, validatorConsumerAddrs := range k.GetAllValidatorsByConsumerAddr(ctx) {
```

Comment:
This getter replaces an iterator where iteration was stopped early. Not stopping early does not impact performance, since the early stopping would only have happened in an error case.